src/libraries/Microsoft.PowerFx.Core/Public/Values/FormulaValueNew.cs

```diff
@@ -38,13 +38,16 @@ public static DecimalValue New(decimal number)
 
         public static DecimalValue New(long number)
         {
-            // it is safe to convert 64-bit Long to Decimal
+            // it is safe to convert 64-bit Long to Decimal, but not to Float
             return new DecimalValue(IRContext.NotInSource(FormulaType.Decimal), (decimal)number);
         }
 
-        public static NumberValue New(int number)
-        {
-            return new NumberValue(IRContext.NotInSource(FormulaType.Number), number);
+        public static DecimalValue New(int number)
```

Comment:
We're hitting issues trying to integrate upstack - _changing return type is a breaking change_. So this leads to binary breaking changes and is blocking a DV update. 

@LucGenetier - need to determine why our API compat checks in PR didn't flag this. 

(Nice explanation here https://stackoverflow.com/questions/15362424/method-with-same-name-and-signature-but-different-return-type-in-c-sharp)